### DISCUSSION TITLE
Ext4.vhdx file cannot be accessed

### DISCUSSION THREAD

**Discussion by cool-delete** [microsoft/WSL#9309] [Q&A]
After rebuild windows, I found the ext4.vhdx of the old system but could not mount it normally. The disk manager prompted me to format it. I used the 7zip program to check the dir and saw two img.

I have a very important file. What method can I use to access the files in it?

![image](https://user-images.githubusercontent.com/13638610/206856673-74364b9d-2bbe-44ee-8113-bee2e4d22d93.png)

Use `WSL -Mount" d: \ ubuntu \ ubuntu_2004.2021.825.0_x64 \ ext4.vhdx "` command

Get Code
Error code: Wsl/Service/AttachDisk/EnumerateDistros/ERROR_PATH_NOT_FOUND

**Comment by Biswa96** [comment]
It looks like the ext4.vhdx file has been formatted to Windows partitions. It can not be used as a distribution in WSL2. Also the ext4.vhdx file should not be mount using disk manager.

**Reply by cool-delete** [reply]
When is it formatting EXT4.VHDX? 
I use the disk manager to load the EXT4.VHDX, which prompts me to format it, but I chose to negate
I use the Linux_Reader.exe program to find some readable format fragments.
 If it is formatted, it should not be seen by the data?

How can I do to read the data I want?

**Reply by Biswa96** [reply]
> I use the Linux_Reader.exe program to find some readable format fragments.

I am not sure what that program does. Formatting a partition does not delete all the data. The Linux_Reader.exe program may have catch those files.

> How can I do to read the data I want?

There are various tools to recover files but I am not sure if those will work. You can find those tools with a little bit of searching.

**Reply by cool-delete** [reply]
For example, do you know similar tools?Can you say the keyword?

**Reply by Biswa96** [reply]
My apology, I wish I could help. I am not familiar with that kind of tools.

**Reply by cool-delete** [reply]
Don't apology, thank you for your answer

**Reply by cool-delete** [reply]
Use the LinuxRead program to mount 1.basic data partition.img to view a part of the file displayed in raw

![image](https://user-images.githubusercontent.com/13638610/207262718-e2bb7e1a-6dc4-4a08-be2f-db8a02f36167.png)

any way to view the directory structure?
